Sources/Metaplex/Programs/TokenMetadata/Accounts/MetadataAccount.swift

```diff
@@ -103,9 +103,25 @@ public struct MetadataAccount: BufferLayout {
         try data.serialize(to: &writer)
         try primarySaleHappened.serialize(to: &writer)
         try isMutable.serialize(to: &writer)
-        try editionNonce?.serialize(to: &writer)
-        try tokenStandard?.rawValue.serialize(to: &writer)
-        try collection?.serialize(to: &writer)
+
+        let hasEditionNonce = editionNonce != nil
+        try hasEditionNonce.serialize(to: &writer)
+        if hasEditionNonce { try editionNonce?.serialize(to: &writer) }
+
+        if let tokenStandard = tokenStandard {
+            try tokenStandard.rawValue.serialize(to: &writer)
+        } else {
+            try UInt8(0).serialize(to: &writer)
+        }
+
+        let hasCollection = collection != nil
+        try hasCollection.serialize(to: &writer)
+        if hasCollection { try collection?.serialize(to: &writer) }
+
+        let padding = MetadataAccount.BUFFER_LENGTH - UInt64(writer.bytes.count)
+        try (0..<padding).forEach { _ in
+            try UInt8(0).serialize(to: &writer)
+        }
```

Comment:
I do not like this way of padding `0` into the data, still investigating best way to do this. Let me know if you have a suggestion.